### ISSUE TITLE
About error

### ISSUE THREAD

**Issue by hao416** [THU-MIG/yoloe#34]
Hello authors, I followed your instructions, but the generated annotation file is throwing an error. Could you help resolve this? The error message is as follows:

WARNING ⚠️ No labels found in /app/shared/datasets/obj365v1/Objects365_v1/train.cache, training may not work correctly. See https://docs.ultralytics.com/datasets for dataset formatting guidance.

**Comment by jameslahm** [comment]
Thanks! 
1. We thought that validating on the LVIS dataset is not related to Objects365v1.
2. COCO dataset metrics are reported with text where the category names of COCO are provided as text prompts.

**Comment by hao416** [comment]
Thank you for your response. Regarding the first question, I would like to clarify whether the LVIS dataset requires preprocessing using other python scripts. Specifically, the lvis.yaml configuration file necessitates defining paths such as train.txt and val.txt, which are not inherently included in the original dataset.

**Comment by jameslahm** [comment]
Hi, LVIS dataset does not require preprocessing and it will be automatically downloaded.

**Comment by hao416** [comment]
> Hi, LVIS dataset does not require preprocessing and it will be automatically downloaded.

ok，thank you very much

**Comment by hao416** [comment]
Hello author, I have an additional question. As I only require detection functionality without segmentation, is it still necessary to execute the tools/generate_sam_masks.py script?

**Comment by jameslahm** [comment]
Hi, no.

**Comment by hao416** [comment]
thank you



---Original---
From: "Wang ***@***.***&gt;
Date: Sat, Mar 22, 2025 14:26 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [THU-MIG/yoloe] About error (Issue #34)


   
Hi, no.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***&gt;
  jameslahm left a comment (THU-MIG/yoloe#34)
 
Hi, no.

**Comment by hao416** [comment]
Hello author, I have a question regarding the COCO dataset metrics. Does the APm metric refer to mAP@50:95? And what does APb specifically indicate?

**Comment by jameslahm** [comment]
Hi, APb indicates box AP@50:95 and APm indicates mask AP@50:95.

**Comment by hao416** [comment]
> Hi, APb indicates box AP@50:95 and APm indicates mask AP@50:95.

ok，thank you！

**Comment by hao416** [comment]
Hello author, I would like to know how to perform visual prompt validation on the LIVS dataset. The code automatically downloaded the LIVS dataset, but an error occurred during validation. Could you please provide guidance on resolving this issue? I have included a screenshot below for reference. This is my first time using the YOLOv8 and YOLOv11 models, and I apologize for not being very familiar with them.

![Image](https://github.com/user-attachments/assets/b329abff-92b4-4c1e-8fa5-208da8748deb)

**Comment by hao416** [comment]
Dear Authors,

  I hope this message finds you well. I am currently working on training the model on my own dataset based on your provided checkpoint, and I would greatly appreciate your guidance on a few questions:
  1.Should I make modifications to train_seg.py or train_pe.py for training on a custom dataset for better performance? （I will load pretrain checkpoints）
  2.Could you please clarify the purpose of the line torch.save({"names": names, "pe": tpe}, pe_path)? Additionally, is last.pt the final checkpoint used for inference?
  3.I noticed that the YAML configuration file used in train_pe.py (e.g., coco.yaml) expects both the training and validation datasets to come from the same source. In my case, I would like to use my own dataset for training and the LVIS minival set for validation. How should I organize the YAML file to support this setup?
  Thank you very much for your time and assistance.

**Comment by Vikashkmrag** [comment]
@hao416 how have you fixed this? I am getting same error for my custom data.
"""
Hello authors, I followed your instructions, but the generated annotation file is throwing an error. Could you help resolve this? The error message is as follows:

WARNING ⚠️ No labels found in /app/shared/datasets/obj365v1/Objects365_v1/train.cache, training may not work correctly. See https://docs.ultralytics.com/datasets for dataset formatting guidance.
"""


Thi is my labels 0th index,
>>> Debug: Processed 9953 images for data/val_segm.json.
>>> Debug: Structure of the first label entry:
  im_file: data/val/d8f50ba4-8bf9-491a-bd13-ff67fc712b46_0.jpg
  shape: (1040, 776)
  cls shape: (126, 1)
  bboxes shape: (126, 4)
  num segments: 126

**Comment by hao416** [comment]
> [@hao416](https://github.com/hao416) how have you fixed this? I am getting same error for my custom data. """ Hello authors, I followed your instructions, but the generated annotation file is throwing an error. Could you help resolve this? The error message is as follows:
> 
> WARNING ⚠️ No labels found in /app/shared/datasets/obj365v1/Objects365_v1/train.cache, training may not work correctly. See https://docs.ultralytics.com/datasets for dataset formatting guidance. """
> 
> Thi is my labels 0th index,
> 
> > > > Debug: Processed 9953 images for data/val_segm.json.
> > > > Debug: Structure of the first label entry:
> > > > im_file: data/val/d8f50ba4-8bf9-491a-bd13-ff67fc712b46_0.jpg
> > > > shape: (1040, 776)
> > > > cls shape: (126, 1)
> > > > bboxes shape: (126, 4)
> > > > num segments: 126

you need to debug the code and organize your dataset in correct format. This error means the
code can not find your json
